pkg/imgpkg/bundle/images_lock.go

```diff
@@ -87,67 +82,13 @@ func (o *ImagesLock) LocalizeImagesLock() (lockconfig.ImagesLock, bool, error) {
 	}
 
 	imagesLock.Images = imageRefs
-	return imagesLock, false, nil
-}
-
-func (o *ImagesLock) LocationPrunedImageRefs(concurrency int) ([]lockconfig.ImageRef, error) {
-	var imageRefs []lockconfig.ImageRef
-
-	errChan := make(chan error, len(o.ImageRefs()))
-	throttle := util.NewThrottle(concurrency)
-	mutex := &sync.Mutex{}
-
-	for _, imgRef := range o.ImageRefs() {
-		newImgRef := imgRef.DeepCopy()
-		go func() {
-			throttle.Take()
-			defer throttle.Done()
-
-			foundImg, err := o.checkImagesExist(newImgRef.Locations())
-			if err != nil {
-				errChan <- err
-				return
-			}
-
-			newImgRef.DiscardLocationsExcept(foundImg)
-
-			mutex.Lock()
-			defer mutex.Unlock()
-
-			imageRefs = append(imageRefs, newImgRef)
-			errChan <- nil
-		}()
-	}
-
-	for range o.ImageRefs() {
-		err := <-errChan
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	return imageRefs, nil
-}
-
-func (o *ImagesLock) checkImagesExist(urls []string) (string, error) {
-	var err error
-	for _, img := range urls {
-		ref, parseErr := regname.NewDigest(img)
-		if parseErr != nil {
-			return "", parseErr
-		}
-		_, err = o.imgRetriever.Digest(ref)
-		if err == nil {
-			return img, nil
-		}
-	}
-	return "", fmt.Errorf("Checking image existence: %s", err)
+	return imagesLock, skippedLocalization, nil
 }
 
-func (o *ImagesLock) imageRelativeToBundle(img string) (string, error) {
+func (o *ImagesLock) imageRelativeToBundle(img string) string {
 	imgParts := strings.Split(img, "@")
 	if len(imgParts) != 2 {
-		return "", fmt.Errorf("Parsing image URL: %s", img)
+		panic(fmt.Sprintf("The provided image URL '%s' does not contain a digest", img))
```

Comment:
throwing in a prefix `Internal inconsistency: ...` will tell user that this is an internal error and not somethign they can fix (aka bug)

Reply:
done